azure-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystem.java

```diff
@@ -489,4 +491,11 @@ Long getPutBlobThreshold() {
     Integer getMaxConcurrencyPerRequest() {
         return this.maxConcurrencyPerRequest;
     }
+
+    public String createSASAppendedURL(String url) throws IllegalStateException {
+        if(Objects.isNull(currentActiveSasCredential)) {
+            throw new IllegalStateException("No current active SAS credential present");
+        }
+        return url + "?" + currentActiveSasCredential.getSignature();
```

Comment:
Is this supposed to look like
```
https://example.com?credential
```
or
```
https://example.com?key=credential
```

Reply:
Nope, you just simply append the credential to the end of a URI
https://learn.microsoft.com/en-us/azure/storage/common/storage-sas-overview#how-a-shared-access-signature-works

The format's new to me too, I had the same exact reaction when I saw that.

Reply:
Sorry, misread your comment. It's the first option so it'l look like...
`https://example.com?sv=...`

...where the `sv=...` portion is the Azure SAS token

Reply:
Makes sense, appreciate the link!